app/src/main/java/com/globant/example/mentorapp/subscriberDetails/presentation/view/fragment/UserDetailsFragment.java

```diff
@@ -1,29 +1,192 @@
+        if (detailsViewModel.getUserRepos() != null) {
+            model.setUserRepos(detailsViewModel.getUserRepos());
+            totalRepositories.setText(
+                    String.format(getActivity().getString(R.string.tv_detail_repositories), detailsViewModel.getUserRepos().size()));
+            usersRepositoryListReady(listReposRecyclerView, listRepoAdapter);
+        }
+        if (detailsViewModel.getError() != null) {
+            switch (detailsViewModel.getError()) {
+                case ERROR_CONNECTION:
+                    showErrorMessage(getString(R.string.communication_error_message));
+                    break;
+                case ERROR_RESPONSE:
+                    showErrorMessage(getString(R.string.http_error_message));
+                    break;
+            }
+        }
+        if (detailsViewModel.getProgress()) {
+            parent.showProgress();
+        } else {
+            parent.hideProgress();
+        }
+    }
+
+    private void subscribeRepoList() {
+        final Observer<List<RepositoryModel>> usersObserver = new Observer<List<RepositoryModel>>() {
+            @Override
+            public void onChanged(@Nullable List<RepositoryModel> repositoryViewModelList) {
+                listRepoAdapter = new UserRepositoriesAdapter(repositoryViewModelList);
+                usersRepositoryListReady(listReposRecyclerView, listRepoAdapter);
+            }
+        };
+        model.getUserRepos().observe(this, usersObserver);
     }
 
+    private void subscribeUserDetails() {
+        final Observer<UserDetailsModel> usersObserver = new Observer<UserDetailsModel>() {
+            @Override
+            public void onChanged(@Nullable UserDetailsModel detailsModel) {
+                setUserInformation(detailsModel);
+            }
+        };
+        model.getUserDetails().observe(this, usersObserver);
+    }
+
+    private void setUserInformation(UserDetailsModel userDetails) {
+        userName.setText(
+                String.format(getActivity().getString(R.string.tv_detail_user_name), userDetails.getUserName()));
+        userLocation.setText(
+                String.format(getActivity().getString(R.string.tv_detail_location), userDetails.getLocation()));
+        company.setText(
+                String.format(getActivity().getString(R.string.tv_detail_company), userDetails.getCompany()));
+        followers.setText(
+                String.format(getActivity().getString(R.string.tv_detail_followers), userDetails.getFollowers()));
+        following.setText(
+                String.format(getActivity().getString(R.string.tv_detail_following), userDetails.getFollowing()));
+
+        Picasso.with(getActivity()).load(userDetails.getImageUrl())
+                .error(R.drawable.ic_not_found)
+                .placeholder(R.drawable.ic_placeholder)
+                .into(userImageView);
+    }
+
+    private void usersRepositoryListReady(RecyclerView recyclerView, RecyclerView.Adapter adapter) {
```

Comment:
If these attributes are part of the class, why are you passing them as parameters?